sos/report/plugins/ceph_mgr.py

```diff
@@ -73,16 +53,26 @@ def setup(self):
             "orch ps",
             "orch status --detail",
             "orch upgrade status",
-            "log last cephadm"
+            "log last cephadm",
+            "mgr dump",
+            "mgr metadata",
+            "mgr module ls",
+            "mgr stat",
+            "mgr versions"
         ])
 
-        self.add_cmd_output(
-            [f"ceph {cmd}" for cmd in ceph_mgr_cmds])
-        # get ceph_cmds again as json for easier automation parsing
-        self.add_cmd_output(
-            [f"ceph {cmd} --format json-pretty" for cmd in ceph_mgr_cmds],
-            subdir="json_output",
-        )
+        # if no orchestraor is configured get rid of the orch stuff
+        orch_needed = self.exec_cmd('ceph orch status')
+        if 'ENOENT' in orch_needed['output']:
```

Comment:
Cirrus tasks report here uncaught exception:

    Traceback (most recent call last):
      File "/tmp/cirrus-ci-build/sos/report/__init__.py", line 1232, in setup
        plug.setup()
      File "/tmp/cirrus-ci-build/sos/report/plugins/ceph_mgr.py", line 66, in setup
        if 'ENOENT' in orch_needed['output']:
    TypeError: a bytes-like object is required, not 'str'

Reply:
I'll change the code but the issue is that command seems to return 2 not 0 

~~~
root@juju-c97b0f-seg-test-2:/home/ubuntu/sosreport/sos# ceph orch status
Error ENOENT: No orchestrator configured (try `ceph orch set backend`)
root@juju-c97b0f-seg-test-2:/home/ubuntu/sosreport/sos# echo $?
2
~~~
So I'll use if orch_needed['status'] == 2 and 'ENOENT' in orch_needed['output']:

does that sound reasonable?

Reply:
What I don't understand is, if there's no ceph, why is ceph_mgr plugin being triggered at all?

Reply:
> I'll change the code but the issue is that command seems to return 2 not 0
> 
> ```
> root@juju-c97b0f-seg-test-2:/home/ubuntu/sosreport/sos# ceph orch status
> Error ENOENT: No orchestrator configured (try `ceph orch set backend`)
> root@juju-c97b0f-seg-test-2:/home/ubuntu/sosreport/sos# echo $?
> 2
> ```
> 
> So I'll use if orch_needed['status'] == 2 and 'ENOENT' in orch_needed['output']:
> 
> does that sound reasonable?

I'd suggest reversing the logic, so that we don't rely on specific return code (2) or the ENOENT errno:
```
      # more commands to be added later
        ceph_mgr_cmds = ([
            "balancer status",
            "log last cephadm",
            "mgr dump",
            "mgr metadata",
            "mgr module ls",
            "mgr stat",
            "mgr versions"
        ])

        # if orchestraor is configured, collect data
        orch_needed = self.exec_cmd('ceph orch status')
        if orch_needed['status'] == 0:
            ceph_mgr_cmds = ([        
                "orch host ls",
                "orch device ls",
                "orch ls",
                "orch ls --export",
                "orch ps",
                "orch status --detail",
                "orch upgrade status"])
```

which would also be cleaner. For example, if a new mgr cmds needs to be added, it needs to be added in two place right now.

Reply:
do you mean to extend the list of strings in the orchestrator configured section. Otherwise I think we end up overwriting ceph_mgt_cmds. For now I've submitted a PR where I have the larger list inside. Can extend it by the orch ones though, that would work too.